### ISSUE TITLE
How to cancel stop market order

### ISSUE THREAD

**Issue by Hyeokreal** [nautechsystems/nautilus_trader#419]
Hi, I'm testing live trading on binance future market. 

I'm facing an issue regrading canceling submitted orders. 
Whenever I set entry order, I also set the stop loss order with stop market. 
However, when my exit strategy closes the position with the profit, the stop market order is leaved alone. 
I think it can affect to other trades opened later.

So, I'm finding the way to cancel the submitted stop market order, but I have no idea.
Can you give me a clue? 

Thank you so much.

**Comment by cjdsellers** [comment]
Hi @Hyeokreal 

This would be because `BracketOrder` is not yet implemented for `CCXT` and the logic is also broken for backtesting too.
This is actually near the top of my list for things to work on.

I'll open a specific issue to track this.

**Comment by cjdsellers** [comment]
Actually we won't be able to get `BracketOrder` working for `CCXT` either, because its not part of their supported unified API. Unless you can suggest a way to 'hack' into the raw params being sent to Binance.

So you know the next integrations we're working on in order are:
- Interactive Brokers
- Binance

And we'll definitely be implementing the `BracketOrder` for both (all) integrations.

**Comment by Hyeokreal** [comment]
I got it. 
I have one more question.

Isn't it available to set reduce_only params for the MarketOrder?
Sometimes, when there are un-cached orders by mistake, market orders for closing position become entry of short position without reduce_only params.

Thank you so much for quick reply.

**Comment by cjdsellers** [comment]
This is an interesting question.

We could support it internally on the Nautilus side, however most APIs don't support reduce-only market orders. So unsure if it makes sense to add that as an option to the `MarketOrder` object or not. 

I've added a note about this and will give it some thought.

**Comment by Hyeokreal** [comment]
I got it. I'll manually add to cancel existing order before closing the position in this stage.

I encountered new issue with the stop market order.
When the stop market order for the stop loss executed in binance future, I got this kinds of message.
I have no idea why stop loss execution occured order expired message.
Do you have any idea to resolve this issue ? 
```
2021-08-20T06:01:50.241070Z [INF] TESTER-001.LiveSkeleton-001: <--[EVT] OrderExpired(instrument_id=OMG/USDT.BINANCE, client_order_id=O-20210820-060000-001-001-2, venue_order_id=11711020232, account_id=BINANCE-001, ts_expired_ns=1629439310240640243).
2021-08-20T06:01:50.241900Z [ERR] TESTER-001.ExecEngine: InvalidStateTrigger(EXPIRED -> ACCEPTED)
 Traceback (most recent call last):
  File "nautilus_trader/execution/engine.pyx", line 596, in nautilus_trader.execution.engine.ExecutionEngine._handle_order_event
    order.apply(event)
  File "nautilus_trader/model/orders/base.pyx", line 578, in nautilus_trader.model.orders.base.Order.apply
    self._fsm.trigger(OrderState.ACCEPTED)
  File "nautilus_trader/core/fsm.pyx", line 117, in nautilus_trader.core.fsm.FiniteStateMachine.trigger
    raise InvalidStateTrigger(f"{self.state_string_c()} -> {self._trigger_parser(trigger)}")
```

I'm so sorry if i'm bothering you with trivial questions.

**Comment by cjdsellers** [comment]
Hi @Hyeokreal,

So the only way this would happen is if an `OrderExpired` event is generated from the CCXT order status message parsed from the socket. I've had a look at the Binance API docs and can't see an option for GTD time in force, which is what the `OrderExpired` is meant to represent (when time has reached the GTD). So this is odd behavior really.

The order state transition table is for maintaining valid state transitions. Once an order is expired it is now completed and that should be its final state, so the lines you added probably aren't correct logically - but if it means you can continue trading and aren't experiencing problems then it is a temporary solution.

I'd suggest, if you're able, to figure out exactly what order status message is coming in from Binance by uncommenting `nautilus_trader/adapters/ccxt/execution.pyx` L506 and having a look at both the CCXT component and the raw `info` part from Binance. You'll see the log line highlighted in green so it stands out, you can change this color too. 

Basically I don't think the order should be getting expired from any of these messages.

Hope that helps and we can track this down!

**Comment by Hyeokreal** [comment]
Thank you so much for your reply!

I found another issue today, but I'll post it with a new issue for other users.
